lms/djangoapps/grades/models.py

```
@@ -0,0 +1,234 @@
+        visible_blocks_model = VisibleBlocks.create(blocks=kwargs['visible_blocks'])
+
+        model = cls.objects.create(
+            user_id=kwargs['user_id'],
+            course_id=kwargs['usage_key'].course_key,
+            usage_key=kwargs['usage_key'],
+            course_version=kwargs['course_version'],
+            subtree_edited_date=kwargs['subtree_edited_date'],
+            earned_all=kwargs['earned_all'],
+            possible_all=kwargs['possible_all'],
+            earned_graded=kwargs['earned_graded'],
+            possible_graded=kwargs['possible_graded'],
+            visible_blocks=visible_blocks_model,
+        )
+        return model
+
+    @classmethod
+    def read(cls, **kwargs):
+        """
+        Reads a grade from database
+
+        Arguments:
+            user_id: The user associated with the desired grade
+            usage_key: The location of the subsection associated with the desired grade
+
+        Raises PersistentSubsectionGrade.DoesNotExist if applicable
+        """
+        return cls.objects.get(
+            user_id=kwargs["user_id"],
+            usage_key=kwargs["usage_key"],
+        )
+
+    @classmethod
+    def update(cls, **kwargs):
+        """
+        Updates a previously existing grade.
+
+        Requires all the arguments listed in docstring for create_grade
+        """
+        grade = cls.objects.get(
+            user_id=kwargs["user_id"],
+            usage_key=kwargs["usage_key"],
+        )
+
+        visible_blocks_model = VisibleBlocks.create(blocks=kwargs['visible_blocks'])
+
+        grade.course_version = kwargs["course_version"]
+        grade.subtree_edited_date = kwargs["subtree_edited_date"]
+        grade.earned_all = kwargs["earned_all"]
+        grade.possible_all = kwargs["possible_all"]
+        grade.earned_graded = kwargs["earned_graded"]
+        grade.possible_graded = kwargs["possible_graded"]
+        grade.visible_blocks = visible_blocks_model
+        grade.save()
```

typos in key names will get silently ignored.  Again, recommend not using kwargs.